### PR TITLE
add an analysis options config

### PR DESCRIPTION
**Pull request by devoncarew**
- add an analysis options config

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/dart_model/analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+include: package:lints/recommended.yaml
```
Comment (jakemac53) [inline pkgs/dart_model/analysis_options.yaml:1]: lets go ahead and use the dart_flutter_team_lints?

Comment (devoncarew) [reply pkgs/dart_model/analysis_options.yaml:1]: I'm fine either way - I just wanted to upgrade this from not using a config

Comment (devoncarew) [reply pkgs/dart_model/analysis_options.yaml]: updated!

**Review by davidmorgan — APPROVED** [review]
Looks good, thanks! I'll merge so you don't end up fixing the next issues too.

**Pull request merged**